### PR TITLE
Add data-line attrs support for heading and paragraph

### PR DESCRIPTION
**Pull request by sofiushko**

### CODE REVIEW

File: src/extensions/base/BaseSchema/BaseSchemaSpecs/index.ts [hunk 1]
```diff
@@ -38,11 +40,18 @@ export const BaseSchemaSpecs: ExtensionAuto<BaseSchemaSpecsOptions> = (builder,
...
                 group: 'block',
                 parseDOM: [{tag: 'p'}],
-                toDOM() {
-                    return ['p', 0];
+                toDOM(node) {
+                    const lineNumber = node.attrs[headingLineNumberAttr];
+
+                    return [
+                        'p',
+                        lineNumber === undefined ? {} : {headingLineNumberAttr: lineNumber},
```
Comment (d3m1d0v) [inline src/extensions/base/BaseSchema/BaseSchemaSpecs/index.ts:52]: > `{default: null}`

so maybe check should be `lineNumber === null` ?

> `{headingLineNumberAttr: lineNumber}`

square brackets is missed: `{[headingLineNumberAttr]: lineNumber}`

File: src/extensions/base/BaseSchema/BaseSchemaSpecs/index.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ export enum BaseNode {
     Paragraph = 'paragraph',
 }
 
+const headingLineNumberAttr = 'data-line';
```
Comment (d3m1d0v) [inline src/extensions/base/BaseSchema/BaseSchemaSpecs/index.ts:11]: > `headingLineNumberAttr`

it's paragraph's attribute name)

File: src/extensions/markdown/Heading/HeadingSpecs/index.ts [hunk 3]
```diff
@@ -31,7 +32,13 @@ export const HeadingSpecs: ExtensionAuto<HeadingSpecsOptions> = (builder, opts)
                 {tag: 'h6', attrs: {[headingLevelAttr]: 6}},
             ],
             toDOM(node) {
-                return ['h' + node.attrs[headingLevelAttr], 0];
+                const lineNumber = node.attrs[headingLineNumberAttr];
+
+                return [
+                    'h' + node.attrs[headingLevelAttr],
+                    lineNumber === undefined ? {} : {headingLineNumberAttr: lineNumber},
```
Comment (d3m1d0v) [inline src/extensions/markdown/Heading/HeadingSpecs/index.ts:39]: same

**Review by d3m1d0v — APPROVED** [review]

**Comment by sofiushko** [comment]
https://github.com/yandex-cloud/yfm-editor/pull/158

**Pull request closed without merging**